proposal/AilurusUmbra/README.md

```diff
@@ -0,0 +1,136 @@
+Most of neural networks contains convolution filters, including EEGNet. 
+Convolution filters could be considered as a high dimensional tensor, so that tensor decomposition (e.g. tucker, CP decomposition) could be applied on each convolution filter to get an approximation of smaller tensors 
+
+* Initialization and Normalization [11, 12, 13, 14]
+
+   Training deep neural networks could be hard when it get deeper, which called degradation problem. This is solved by adding shortcut connection and batch normaliztion in the design of design ResNet [14]. Some works [12, 13] have succeeded in training very deep networks without batch normalization, while these papers required othogonal initialization. Different initialization or normalization tricks are all trying to make some hypotheses on data or dynamics weights, which are based on some probabily distribution or statistics independent.  A few weeks ago, DeepMind proposed a new technique called SkipInit [11] to replace different kinds of initialization or normalization, and they also gave an explanation of how normalization works. As [11] mentioned, although BatchNorm is very widely used in nowadays neural network design, this would break statistical independent between each batch. This project would try to implement SkipInit to replace BatchNorm in EEGNet.
+
+* Optimization 
+  
+   In recent years, there are tons of optimizer proposed for different kind of NNs. However, it is not too sensitive to choose fancy optimizer for training light-weighted NN due to the short backpropagation path, so this project does not pay too much effort here.
+
+* Depthwise Separable Convolution [15, 16]
+
+   From aspects of Linear Algebra, we could do tensor decomposition on each of convolution filter to get an approximation of smaller filters. Depthwise Separable Convolution [15] is to use 2 smaller layers to approximate a larger convolution layer [Figure 3.]. This tricks are widely used in small scale CNN.
+![](https://miro.medium.com/max/1200/1*e_oU-f6hX4FSSD-1OukP6Q.png) Figure 3. [16]
+
+* Convolution filter size [17]
+  
+  More and more papers of Neural Architecture Search show that 5x5 convolution filter might give us better result than 3x3 filters [17]. This project would compare results of both 5x5 and 3x3 convolution filters.
+
+### Numerical Method to be Applied
+
+* Initialization and Normalization
+  * BatchNorm with random initialization
+  * SkipInit
+  
+* Optimization 
+  * SGD
+  * Adagrad
+  * Adam
+
+* Convolution Filter
+  * 5x5 window 
+  * 3x3 window
+  * 5x5 depthwise-separable
+  * 3x3 depthwise separable
+ 
+
+## Perspective users
+
+* BCI researchers for processing EEG/MEG.
+* Engineers that need to do multichannel time series prediction/classification.
+
+## System architecture
+
+Most of the computation would be leverage to Pytorch framework, so that tensors could be easily move to VRAM of GPU to compute, and move back to RAM after computation.
```

Comment:
The course expects you to write C++ code on CPU.  If you write GPU code, or use only Pytorch Python API, your do not implement C++/Python interface yourselves, and do not fulfill the expectation.

You can try to address the omission, but I am not sure what is the best way to do it.  If the omission remains unaddressed at the end of the project, your grade will be affected.

Reply:
I think I should take some time to change this proposal into mainly C++ project (maybe compatible with PyTorch). I would commit this later.

Reply:
I remove the GPU part in this commit [e07643d](https://github.com/AilurusUmbra/nsdhw_20sp/commit/e07643d3e677a07449128c375017a2c6f70c133c)